src/components/About/AboutUIModal.tsx

```diff
@@ -1,24 +1,71 @@
 import * as React from 'react';
-import { AboutModal } from 'patternfly-react';
+import { AboutModal, Spinner } from 'patternfly-react';
+import * as API from '../../services/Api';
+import update from 'immutability-helper';
 
 const pfLogo = require('../../img/logo-alt.svg');
+const SWS_CORE_COMMIT_HASH = 'SWS core commit hash';
+const SWS_CORE_VERSION = 'SWS core version';
 
 type AboutUIModalState = {
   showModal: boolean;
+  loadingVersions: boolean;
+  versions: Array<AboutUIModalService>;
+};
+
+type AboutUIModalService = {
+  name: string;
+  version: string;
+};
+
+const getStatus = () => {
+  return API.GetStatus().then(response => {
+    const rawStatus = response['data'];
+    return [
+      {
+        name: 'swscore',
+        version: `${rawStatus[SWS_CORE_VERSION]} (${rawStatus[SWS_CORE_COMMIT_HASH]})`
+      }
+    ];
+  });
 };
 
 class AboutUIModal extends React.Component<Object, AboutUIModalState> {
   constructor(props: any) {
     super(props);
-    this.state = { showModal: false };
+    this.state = { showModal: false, loadingVersions: false, versions: [] };
   }
 
   open = () => {
-    this.setState({ showModal: true });
+    this.setState(state => {
+      if (!state.loadingVersions) {
+        getStatus().then(
+          status => {
+            this.setState(innerState => {
+              return update(innerState, {
+                loadingVersions: { $set: false },
+                versions: { $set: status }
+              });
```

Comment:
what's the difference between this and:

```
           this.setState({
               loadingVersions: false,
               versions: status
             });
```
?

In other words I have the impression that react already uses an immutable state, no?

Reply:
I'm telling:
Set loadingVersions to false and versions to status, but don't touch anything else.
e.g. if showModal = true, it will stay as true, if is false, it will stay as false.

I could of course just use 
```javascript
 this.setState({
               loadingVersions: false,
               versions: status,
               showModal: innerState.showModal
             });
```
But this will start getting complicated if i have to keep track of everything in the state for a single update. 
e.g. If i had a new attribute, I would have to track every `setState` and update accordingly. Type safety helps, but if an attribute is optional, it won't warn if i accidentally remove it.

Reply:
I don't think react's `setState` is a dumb setter, I think it already merges the input object with the existing one so you don't have to forward all the unmodified properties.

... or else the code I wrote til now is completely bugged and just work by chance? :-)

Reply:
I though of it as a dumb setter :). 

I'll check more about it and report back, i got this usage from https://github.com/kolodny/immutability-helper

Reply:
`This performs a shallow merge of stateChange into the new state`

I'll update my code.